x-pack/plugins/maps/public/classes/fields/agg/percentile_agg_field.ts

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { IndexPattern } from 'src/plugins/data/common/index_patterns/index_patterns';
+import { AGG_TYPE } from '../../../../common/constants';
+import { IESAggField, CountAggFieldParams } from './agg_field_types';
+import { CountAggField } from './count_agg_field';
+import { addFieldToDSL, getField } from '../../../../common/elasticsearch_util';
+import { ESDocField } from '../es_doc_field';
+import { getOrdinalSuffix } from '../../util/ordinal_suffix';
+
+export interface PercentileAggParams extends CountAggFieldParams {
+  esDocField?: ESDocField;
+  percentile: number;
+}
+
+export class PercentileAggField extends CountAggField implements IESAggField {
```

Comment:
Would it be better to extend `AggField` instead of `CountAggField`? That would remove a lot of the duplicated methods that are the same as `AggField`.